### PR TITLE
[threads] Rely on MonoThreadInfo.stack_start_limit and MonoThreadInfo.stack_end instead of calling mono_thread_info_get_stack_bounds which is very slow

### PR DESCRIPTION
**Pull request by luhenry**
Fixes https://bugzilla.xamarin.com/show_bug.cgi?id=58911

### CODE REVIEW

File: mono/metadata/icall.c [hunk 1]
```diff
@@ -1017,33 +1017,24 @@ ves_icall_System_Runtime_CompilerServices_RuntimeHelpers_SufficientExecutionStac
...
-	current = (guint8 *)&stack_addr;
-	if (current > stack_addr) {
-		if ((current - stack_addr) < min_size)
-			return FALSE;
-	} else {
-		if (current - (stack_addr - stack_size) < min_size)
-			return FALSE;
-	}
+	/* .net seems to check that at least 50% of stack is available */
+	if (stack_current < stack_start + stack_size / 2)
```
Comment (kumpera) [inline mono/metadata/icall.c]: This is wrong, dotnet checks for a specific amount of bytes available.

File: mono/metadata/icall.c [hunk 2]
```diff
@@ -1017,33 +1017,24 @@ ves_icall_System_Runtime_CompilerServices_RuntimeHelpers_SufficientExecutionStac
...
 
-	thread = mono_thread_internal_current ();
-	// .net seems to check that at least 50% of stack is available
-	min_size = thread->stack_size / 2;
+	stack_current = (guint8 *)&stack_start;
 
-	// TODO: It's not always set
-	if (!min_size)
+	/* Some platforms support dynamic stack resizing */
+	if (stack_current < stack_start)
```
Comment (kumpera) [inline mono/metadata/icall.c]: This is not correct, even on platforms that support growable stacks (windows) they have a bounded max size and we compute that in mono_thread_info_get_stack_bounds

Comment (vargaz) [reply mono/metadata/icall.c]: The linux implementation is also slow for the main thread at least, i.e.:
https://bugzilla.xamarin.com/show_bug.cgi?id=58911

**Comment by lewurm** [comment]
is this superseded by https://github.com/mono/mono/pull/5543?

The refactoring around `mono_{,arch_}handle_exception` should be a separate PR.

**Pull request closed without merging**